src/test/java/uk/gov/ons/ctp/integration/envoycuc/mockclient/MockClient.java

```diff
@@ -0,0 +1,292 @@
+package uk.gov.ons.ctp.integration.envoycuc.mockclient;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.time.DateUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.ons.ctp.common.domain.CaseType;
+import uk.gov.ons.ctp.common.domain.UniquePropertyReferenceNumber;
+import uk.gov.ons.ctp.integration.common.product.model.Product;
+import uk.gov.ons.ctp.integration.envoycuc.client.RateLimiterClientRequest;
+import uk.gov.ons.ctp.integration.envoycuc.client.TestClient;
+import uk.gov.ons.ctp.integration.ratelimiter.client.RateLimiterClient;
+import uk.gov.ons.ctp.integration.ratelimiter.model.CurrentLimit;
+import uk.gov.ons.ctp.integration.ratelimiter.model.LimitStatus;
+import uk.gov.ons.ctp.integration.ratelimiter.model.RateLimitResponse;
+
+@Data
```

Comment:
Some Javadoc to explain characteristics would help:

- simulates the combination of the client access code + the real envoy limiter
- will store details as each test is run, (is NOT recreated for each test). Normally this would be undesirable in a test situation, but in our case it more realistically models the actual envoy limiter, since we won't be resetting it
- because it changes state for each test run, then care must be taken creating tests that do not use data that has been used before
- the in-memory data structures will grow big very fast, so care must be taken if a LOT of calls are made